tpp-app/src/services/cycle/CycleService.js

```diff
@@ -0,0 +1,290 @@
+    return twoDaysLater;
+}
+
+const CycleService = {
+  /**
+   *  Store how far the user is into their period as a percentage
+   *  @param {number} percent Float in range [0,1] of how far along period is
+   *  @return {Promise} Resolves when the set operation is completed
+   */
+  POSTCycleDonutPercent: async function(percent){
+    try {
+      var today = new Date();
+      var date = getDateString(today);
+
+      let datePercent = {
+      }
+      datePercent[date] = percent;
+      return await AsyncStorage.setItem(Keys.CYCLE_DONUT_PERCENT, JSON.stringify(datePercent));
+    } catch (e) {
+      console.log(e);
+      return null;
+    }
+  },
+
+  /**
+   * Get the user's average period length
+   * @return {Promise} Resolves into either an integer for number of days or NULL if info not present
+   */
+  GETAveragePeriodLength: async function(){
+    try {
+      const res = await AsyncStorage.getItem(Keys.AVERAGE_PERIOD_LENGTH);
+      return res;
+    } catch (e) {
+      console.log(e);
+      return null;
+    }
+
+  },
+  /**
+   * Get the user's average cycle length
+   * @return {Promise} Resolves into either an integer for number of days or NULL if info is not present
+   */
+  GETAverageCycleLength: async function(){
+    try {
+      const res = await AsyncStorage.getItem(Keys.AVERAGE_CYCLE_LENGTH);
+      return res;
+    } catch (e) {
+      console.log(e);
+      return null;
+    }
+
+  },
+
+  /**
+   * Get the number of days the user has been on their period
+   *
+   * @param {Object} calendar The object containing the symptoms for this year, last year, and next year. Optional.
+   * @return {Promise} Resolves into 0 if user not on period, and an integer of the days they have been on their period otherwise
+   */
+  GETPeriodDay: async function (calendar = null){
+
+    let periodDays = 0;
+    var date = new Date()
+    if (!calendar){
+      calendar = await getCalendarByYear(date.getFullYear());
+    }
+    let dateSymptoms = await getSymptomsFromCalendar(calendar, date.getDate(), date.getMonth()+1, date.getFullYear());
+    if (dateSymptoms.flow === null || dateSymptoms.flow === FLOW_LEVEL.NONE){
+      return 0;
+    }
+    else {
+      let startDate = await this.GETMostRecentPeriodStartDate(calendar);
+      return getDaysDiffInclusive(startDate, date);
+    }
+
+    return periodDays;
+
+  },
+
+
+  /**
+   * Get most recent period start date
+   *
+   * @param {Object} calendar The object containing the symptoms for this year, last year, and next year. Optional.
+   * @return {Promise} A promise that resolves into a Date object that is when the most recent period started.
+   */
+  GETMostRecentPeriodStartDate: async function (calendar = null) {
+    var date = new Date()
+
+    if (!calendar){
+      calendar = await getCalendarByYear(date.getFullYear());
+    }
+
+    let mostRecentPeriodDay = getLastPeriodStart(date, calendar);
```

Comment:
If we have a new user who has no periods logged yet, what will happen? I think the while loop in getLastPeriodStart() will be an infinite loop